plugins/modules/ec2_asg_lifecycle_hook.py

```diff
@@ -150,23 +154,30 @@ def create_lifecycle_hook(connection, module):
         module.fail_json_aws(e, msg="Failed to get Lifecycle Hook")
 
     if not existing_hook:
-        changed = True
-    else:
-        # GlobalTimeout is not configurable, but exists in response.
-        # Removing it helps to compare both dicts in order to understand
-        # what changes were done.
-        del(existing_hook[0]['GlobalTimeout'])
-        added, removed, modified, same = dict_compare(lch_params, existing_hook[0])
-        if added or removed or modified:
-            changed = True
-
-    if changed:
         try:
+            return_object['changed'] = True
             connection.put_lifecycle_hook(**lch_params)
+            return_object['lifecycle_hook_info'] = connection.describe_lifecycle_hooks(
+                AutoScalingGroupName=asg_name, LifecycleHookNames=[lch_name])['LifecycleHooks']
         except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
             module.fail_json_aws(e, msg="Failed to create LifecycleHook")
 
-    return(changed)
+    else:
+        added, removed, modified, same = dict_compare(lch_params, existing_hook[0])
+        if modified:
+            # GlobalTimeout is not configurable, but exists in response.
+            # Removing it helps to compare both dicts in order to understand
+            # what changes were done.
+            del(existing_hook[0]['GlobalTimeout'])
+            try:
+                return_object['changed'] = True
+                connection.put_lifecycle_hook(**lch_params)
+                return_object['lifecycle_hook_info'] = connection.describe_lifecycle_hooks(
+                    AutoScalingGroupName=asg_name, LifecycleHookNames=[lch_name])['LifecycleHooks']
+            except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+                module.fail_json_aws(e, msg="Failed to create LifecycleHook")
+
+    module.exit_json(**camel_dict_to_snake_dict(return_object))
```

Comment:
I like that we're returning the object now, but we should to document the values that are returned in the RETURN block of the documentation and include a comment in the changelog that the module now returns information about the LCH, so users can update their playbooks if they want to take advantage of this data.